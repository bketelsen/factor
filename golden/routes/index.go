package routes

import (
	"fmt"
	"strconv"

	"github.com/gowasm/vecty"
)

type Index struct {
	vecty.Core
	Notice string
	count  int
}

func (i *Index) OnClick(e *vecty.Event) {
	fmt.Println("Someone clicked on me", e.Target)
	i.count++
	i.Notice = "Click Count: " + strconv.Itoa(i.count)

	vecty.Rerender(i)
}
