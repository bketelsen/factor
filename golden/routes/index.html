<main>
	<components:Nav MyProp="blue"/>
	<h1>Great success!</h1>

<figure>
	<img alt='Borat' src='/assets/great-success.png'></img>
	<figcaption>HIGH FIVE!</figcaption>
</figure>


<p>
    <input type="submit" value="increment counter" vecty:onClick="OnClick" />
	<strong>{vecty-field:CountText}</strong>
</p>
</main>
<style>
	h1,
	figure,
	p {
		text-align: center;
		margin: 0 auto;
	}

	h1 {
		font-size: 2.8em;
		text-transform: uppercase;
		font-weight: 700;
		margin: 0 0 0.5em 0;
	}

	figure {
		margin: 0 0 1em 0;
	}

	img {
		width: 100%;
		max-width: 400px;
		margin: 0 0 1em 0;
	}

	p {
		margin: 1em auto;
	}

	@media (min-width: 480px) {
		h1 {
			font-size: 4em;
		}
	}
</style>
